subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy

```diff
@@ -19,13 +19,16 @@ package org.gradle.performance.regression.nativeplatform
 import org.gradle.initialization.ParallelismBuildOptions
 import org.gradle.performance.AbstractCrossVersionGradleInternalPerformanceTest
 import org.gradle.performance.mutator.AbstractFileChangeMutator
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import spock.lang.Unroll
 
+@Requires(TestPrecondition.LINUX)
 class SwiftBuildPerformanceTest extends AbstractCrossVersionGradleInternalPerformanceTest {
```

Comment:
Why not run this on macOS too?

Reply:
All scenarios are pretty expensive (~6h on mac agents), so my understanding is that we're going to run only a small number of scenarios on macos. The exact subset of every OS can be done in a followup https://docs.google.com/document/d/1F_m6XLOWvpLdaGJ7ey6sta3mgC-HymAYe-EttNwhMss/edit

Reply:
Let's add some documentation to these exclusions. Maybe we should add an optional `because` parameter to `@Requires`. Or we could just leave a comment here.

I think it would make sense to also add explanations to all the other `@Requires` annotations in this PR.

Reply:
We really should change how we decide which tests to run on which operating systems. I think adding an `@Requires` is OK for tests which don't run on some OS, but it shouldn't be used to decide what to run on which OS.

For this test, I suppose it should be `@Requires(TestPrecondition.NOT_WINDOWS)`.